lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -136,16 +136,24 @@ namespace Js
         Var newVarInstance = nullptr;
         if (callInfo.Flags & CallFlags_New)
         {
-          if (JavascriptProxy::Is(targetFunction))
-          {
-            JavascriptProxy* proxy = JavascriptProxy::FromVar(targetFunction);
-            Arguments proxyArgs(CallInfo(CallFlags_New, 1), &targetFunction);
-            args.Values[0] = newVarInstance = proxy->ConstructorTrap(proxyArgs, scriptContext, 0);
-          }
-          else
-          {
-            args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
-          }
+            if (args.HasNewTarget())
+            {
+                // target has an overriden new target make a new object from the newTarget
+                RecyclableObject* newTarget = RecyclableObject::UnsafeFromVar(args.Values[args.Info.Count]);
```

Comment:
I don't think the double check is a problem; I expect it will go away after inlining and optimization. There are a few other places that do HasNewTarget/GetNewTarget pairs like JavascriptExternalFunction::StdCallExternalFunctionThunk, and I think it looks cleaner and safer that way unless we have strong data showing that it hurts performance.

Reply:
OK - I'll update.